io/driver/module_unload_load.py

```diff
@@ -0,0 +1,139 @@
+# Author: Naresh Bannoth<nbannoth@in.ibm.com>
+#
+
+import time
+import os
+from avocado import main
+from avocado.utils import process
+from avocado.utils import linux_modules, genio
+from avocado.utils import pci
+from avocado import Test
+
+
+class ModuleLoadUnload(Test):
+
+    """
+    This test is for driver module unload and load verification.
+    :param module: Name of the driver module
+    :param iteration: Number of time to unload and load the module
+    :only_io True for single provide module and False for All pci modules
+    """
+    def setUp(self):
+        """
+        get parameters.
+        """
+        self.module = self.params.get('module', default=None).split(',')
+        self.iteration = self.params.get('iteration', default=1)
+        self.only_io = self.params.get('only_io', default=None)
+        self.load_unload_sleep_time = 30
+        self.error_modules = []
+        self.mod_list = []
+        self.uname = linux_modules.platform.uname()[2]
+
+    def built_in_module(self, module):
+        """
+        checking whether the given module is built_in module or not
+        """
+        # self.uname = linux_modules.platform.uname()[2]
+        path = "/lib/modules/%s/modules.builtin" % self.uname
+        for each in genio.read_all_lines(path):
+            out = process.getoutput(each.split('/')[-1])
+            if module == out.split('.'[0]):
+                return True
+            return False
+
+    def get_depend_modules(self, module):
+        """
+        Returns the dependent modules
+        """
+        config_path = os.path.join(os.path.abspath(''),
+                                   "module_unload_load.py.data/config")
+        for line in genio.read_all_lines(config_path):
+            if module in line:
+                return line.split('=')[-1]
+
+    def module_load_unload(self, module_list):
+        """
+        Unloading and loading the given module
+        """
+        for mod1 in module_list:
+            if linux_modules.module_is_loaded(mod1) is False:
+                linux_modules.load_module(mod1)
+                time.sleep(self.load_unload_sleep_time)
+
+        for _ in range(0, self.iteration):
+            for mdl in module_list:
+                sub_mod = self.get_depend_modules(mdl)
+                if sub_mod:
+                    sub_mod = sub_mod.split(' ')
+                    for mod in sub_mod:
+                        if mod == 'multipath':
+                            process.system("multipath -F", ignore_status=True)
+                            cmd = "lsmod | grep -i ^%s" % mdl
+                            for _ in range(20):
+                                time.sleep(3)
+                                out = process.getoutput(cmd).split(" ")[-1]
+                                if out.strip() != '0':
+                                    self.log.info("command value : %s\n" % out)
+                                    process.system("multipath -F",
```

Comment:
why doing `multipath -F` twice ?

Reply:
This block handling multipath can be a separate function. And it can be handled a lot better

Reply:
ok, will make a separate function for this.

Reply:
@Naresh-ibm 
check if this works:
```
      cmd = "lsmod | grep -i ^%s" % mdl
      for i in range(20):
          if process.getoutput(cmd).split(" ")[-1] == '0':
              return True
          process.system("multipath -F", ignore_status=True)
          time.sleep(3)
          continue
      return False
```

Reply:
@Naresh-ibm I would also suggest converting the bash command to pythonic way